app/uk/gov/hmrc/softdrinksindustrylevystub/services/SubscriptionGenerator.scala

```diff
@@ -22,15 +22,152 @@ import cats.implicits._
 import org.scalacheck._
 import org.scalacheck.support.cats._
 import uk.gov.hmrc.smartstub._
-import uk.gov.hmrc.softdrinksindustrylevystub.models._
+import uk.gov.hmrc.softdrinksindustrylevystub.models.internal._
+import uk.gov.hmrc.softdrinksindustrylevystub.models.maxL
+import ActivityType._
+import uk.gov.hmrc.softdrinksindustrylevystub.models.CreateSubscriptionResponse
+import uk.gov.hmrc.softdrinksindustrylevystub.models.EnumUtils.idEnum
 
 object SubscriptionGenerator {
 
+  lazy val store: PersistentGen[String, Option[Subscription]] = genSubscription.asMutable[String]
+
+  def genSubscription: Gen[Option[Subscription]] = {
+    SdilNumberTransformer.tolerantUtr.gen |@|
+      orgNameGen |@|
+      Gen.oneOf("1", "2", "3", "4", "5").almostAlways |@|
+      addressGen |@|
+      internalActivityGen |@|
+      Gen.date(2018, 2028) |@|
+      Gen.choose(0, 10).flatMap(Gen.listOfN(_, siteGen)) |@|
+      Gen.choose(0, 10).flatMap(Gen.listOfN(_, siteGen)) |@|
+      contactGen
+  } map { Subscription.apply } rarely
+
   def genCreateSubscriptionResponse: Gen[CreateSubscriptionResponse] = {
-    Gen.const(LocalDateTime.now.atOffset(ZoneOffset.UTC))    |@| // processingDate
-    pattern"999999999999".gen                                    // formBundleNumber
+    Gen.const(LocalDateTime.now.atOffset(ZoneOffset.UTC)) |@| // processingDate
+      pattern"999999999999".gen // formBundleNumber
   }.map(CreateSubscriptionResponse.apply)
 
+  private lazy val internalActivityGen: Gen[Activity] = for {
+    produced <- activityGen(ProducedOwnBrand).sometimes
+    imported <- activityGen(Imported).sometimes
+    copacking <- activityGen(CopackerAll).sometimes
+    copacked <- activityGen(Copackee).sometimes
+    isLarge <- Gen.boolean
+  } yield {
+    InternalActivity(
+      Seq(produced, imported, copacking, copacked).flatten.toMap,
+      isLarge
+    )
+  }
+
+  private def activityGen(at: ActivityType.Value): Gen[(ActivityType.Value, LitreBands)] = for {
+    l <- Gen.choose(0, maxL)
+    h <- Gen.choose(0, maxL)
+  } yield {
+    at -> (l -> h)
+  }
+
+  private lazy val contactGen: Gen[Contact] = for {
+    fname <- Gen.forename().usually
+    lname <- Gen.surname
+    position <- jobTitleGen.usually
+    phoneNumber <- pattern"9999 999999"
+    email <- genEmail
+  } yield {
+    Contact(fname.map(f => s"$f $lname"), position, "0" + phoneNumber, email)
+  }
+
+  private lazy val genEmail: Gen[String] = for {
+    fname <- Gen.forename()
+    lname <- Gen.surname
+    domain <- Gen.oneOf("gmail", "outlook", "yahoo", "mailinator", "example")
+    tld <- Gen.oneOf("com", "co.uk")
+  } yield {
+    s"$fname.$lname@$domain.$tld"
+  }
+
+  private lazy val siteGen: Gen[Site] = for {
```

Comment:
consider making some of these non-private, we might want to use your generators on their own (for example to get a site).